project/shacl/modos_schema.shacl.ttl

```diff
@@ -126,291 +140,335 @@ modos:VariantSet a sh:NodeShape ;
-        [ sh:datatype xsd:string ;
-            sh:description "A human-readable name for a thing" ;
+            sh:description "A string specifying the release or version of a software or resource." ;
             sh:maxCount 1 ;
-            sh:order 6 ;
-            sh:path schema1:name ],
-        [ sh:description "A unique identifier for a thing" ;
+            sh:nodeKind sh:Literal ;
+            sh:order 4 ;
+            sh:path modos:version ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "A unique identifier for a thing" ;
             sh:maxCount 1 ;
             sh:minCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 5 ;
             sh:path schema1:identifier ],
         [ sh:datatype xsd:string ;
-            sh:description "A string specifying the release or version of a software or resource." ;
+            sh:description "A human-readable description for a thing" ;
             sh:maxCount 1 ;
-            sh:order 4 ;
-            sh:path modos:version ] ;
+            sh:nodeKind sh:Literal ;
+            sh:order 7 ;
+            sh:path schema1:description ] ;
     sh:targetClass modos:ReferenceGenome .
 
 modos:Sample a sh:NodeShape ;
     sh:closed true ;
     sh:description "A biological sample used in assays. Examples include a whole organism, tissue or cell line." ;
     sh:ignoredProperties ( rdf:type ) ;
-    sh:property [ sh:datatype xsd:string ;
-            sh:description "The organization responsible for collecting a given sample." ;
-            sh:order 4 ;
-            sh:path modos:collector ],
-        [ sh:datatype xsd:integer ;
+    sh:property [ sh:datatype xsd:integer ;
             sh:description "The taxid number describing the taxonomic range of a sample." ;
+            sh:nodeKind sh:Literal ;
             sh:order 3 ;
             sh:path modos:taxon_id ],
         [ sh:datatype xsd:string ;
-            sh:description "The cell type name." ;
+            sh:description "The organization responsible for collecting a given sample." ;
+            sh:nodeKind sh:Literal ;
+            sh:order 4 ;
+            sh:path modos:collector ],
+        [ sh:datatype xsd:string ;
+            sh:description "A human-readable description for a thing" ;
             sh:maxCount 1 ;
-            sh:order 0 ;
-            sh:path modos:cell_type ],
+            sh:nodeKind sh:Literal ;
+            sh:order 7 ;
+            sh:path schema1:description ],
         [ sh:datatype xsd:string ;
             sh:description "A human-readable name for a thing" ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 6 ;
             sh:path schema1:name ],
-        [ sh:description "A unique identifier for a thing" ;
+        [ sh:description "The cell type name or code, if applicable." ;
+            sh:maxCount 1 ;
+            sh:or ( [ sh:in [ rdf:rest () ] ] [ sh:datatype xsd:string ;
+                        sh:nodeKind sh:Literal ] ) ;
```

Comment:
sh:Literal redundant if you already restrict to string.

Reply:
also - it will cause 2 errors for 1 value which can give an overly pessimistic view of your data, and a bloated validation report.

Reply:
shit this comment is on the wrong line, but it's not entirely un-relevant: sh:or for sh:nodeKind sh:Literal means someone can put `"0"^^xsd:integer` or `false` as a valid value here for cell type name.

Reply:
this only seems to happen with dynamic enums, so that would be an argument for using a regular field with string or uri value instead